internal/services/signalr/web_pubsub_custom_certificate_resource.go

```diff
@@ -165,11 +165,12 @@ func (r CustomCertWebPubsubResource) Read() sdk.ResourceFunc {
 			if err != nil {
 				return fmt.Errorf("getting key vault base uri from %s: %+v", id, err)
 			}
-			vaultId, err := commonids.ParseKeyVaultID(*keyVaultIdRaw)
-			if err != nil {
-				return fmt.Errorf("parsing key vault %s: %+v", vaultId, err)
+			if keyVaultIdRaw != nil {
+				vaultId, err := commonids.ParseKeyVaultID(*keyVaultIdRaw)
+				if err != nil {
+					return fmt.Errorf("parsing key vault %s: %+v", vaultId, err)
+				}
 			}
```

Comment:
just wanted to check what is the purpose of parsing vaultId here? it doesn't look like we use this value anywhere?

Reply:
Hi @catriona-m thanks for your time. Since I am not the author of this code, I don't know what the purpose of adding vaultId parsing is in the first place. From what I understand, the purpose of this code is to verify the validity of the `KeyVaultBaseUri` returned by the API. Other than that, it really doesn't do anything.

Anyway, I think it is necessary to check whether the pointer type variable `keyVaultIdRaw` is nil before using it. Of course, it is also okay for me to delete this code, what do you think?

Reply:
thanks @sinbai - I think this should be fine 👍